python/packages/autogen-ext/src/autogen_ext/tools/azure/_ai_search.py

```diff
@@ -0,0 +1,1133 @@
+    ) -> ExpectedType: ...
+
+    @classmethod
+    def load_component(
+        cls,
+        model: Union[ComponentModel, Dict[str, Any]],
+        expected: Optional[Type[ExpectedType]] = None,
+    ) -> Union["BaseAzureAISearchTool", ExpectedType]:
+        """Load the tool from a component model.
+
+        Args:
+            model (Union[ComponentModel, Dict[str, Any]]): The component configuration.
+            expected (Optional[Type[ExpectedType]]): Optional component class for deserialization.
+
+        Returns:
+            Union[BaseAzureAISearchTool, ExpectedType]: An instance of the tool.
+
+        Raises:
+            ValueError: If the component configuration is invalid.
+        """
+        if expected is not None and not issubclass(expected, BaseAzureAISearchTool):
+            raise TypeError(f"Cannot create instance of {expected} from AzureAISearchConfig")
+
+        target_class = expected if expected is not None else cls
+        assert hasattr(target_class, "_from_config"), f"{target_class} has no _from_config method"
+
+        if isinstance(model, ComponentModel) and hasattr(model, "config"):
+            config_dict = model.config
+        elif isinstance(model, dict):
+            config_dict = model
+        else:
+            raise ValueError(f"Invalid component configuration: {model}")
+
+        config = AzureAISearchConfig(**config_dict)
+
+        tool = target_class._from_config(config)
+        if expected is None:
+            return tool
+        return cast(ExpectedType, tool)
+
+    def return_value_as_string(self, value: SearchResults) -> str:
+        """Convert the search results to a string representation.
+
+        This method is used to format the search results in a way that's suitable
+        for display to the user or for consumption by language models.
+
+        Args:
+            value (List[SearchResult]): The search results to convert.
+
+        Returns:
+            str: A formatted string representation of the search results.
+        """
+        if not value.results:
+            return "No results found."
+
+        result_strings: List[str] = []
+        for i, result in enumerate(value.results, 1):
+            content_str = ", ".join(f"{k}: {v}" for k, v in result.content.items())
+            result_strings.append(f"Result {i} (Score: {result.score:.2f}): {content_str}")
+
+        return "\n".join(result_strings)
+
+
+_allow_private_constructor = ContextVar("_allow_private_constructor", default=False)
+
+
+class OpenAIAzureAISearchTool(BaseAzureAISearchTool):
+    """A tool for performing searches using Azure AI Search with OpenAI embeddings.
+
+    This tool extends the base Azure AI Search tool with OpenAI embedding capabilities
+    for generating vector embeddings at query time.
+
+    Note:
+        Do not initialize this class directly. Use factory methods like
+        create_semantic_search(), create_vector_search(), or load_component() instead.
+    """
```

Comment:
We need to have API docs with examples here. The examples can be copied and pasted to a script and run.